crates/ra_lsp_server/src/main_loop/handlers.rs

```diff
@@ -514,29 +514,26 @@ pub fn handle_hover(
         Some(it) => it,
     };
     let mut result = Vec::new();
+    let file_id = params.text_document.try_conv_with(&world)?;
+    let file_range = FileRange {
+        file_id,
+        range: rr.reference_range,
+    };
+    if let Some(type_name) = world.analysis().type_of(file_range)? {
```

Comment:
Now we have an opposite problem: if we fail to retrive the type, we won't even try retrieving the docs. These two bits really have to be extracted into two helper functions to be called from `handle_hover`.